src/test/scala/com/exasol/spark/DefaultSourceSuite.scala

```diff
@@ -17,4 +18,13 @@ class DefaultSourceSuite extends FunSuite with Matchers with MockitoSugar {
       thrown.getMessage === "A sql query string should be specified when loading from Exasol"
     )
   }
+
+  // scalastyle: off
+  test("mergeConfiguration should merge runtime sparkconf into user provided configuration") {
+    val sparkConf = Map[String, String]("spark.exasol.username" -> "newUsername")
+    val parameters = Map[String, String]("username" -> "oldUsername", "password" -> "password")
+
+    val newConf = new DefaultSource().mergeConfiguration(parameters, sparkConf)
+    assert(newConf.getOrElse("username", "not available") === "newUsername")
```

Comment:
Hey @3cham @jpizagno 

Thanks for the PR! 

Minor thing, could you also add another assert for checking password. This will show us that not merged params are there from options.